mpqp/execution/providers/aws.py

```diff
@@ -319,3 +319,47 @@ def get_result_from_aws_task_arn(task_arn: str) -> Result:
     device = AWSDevice.from_arn(device_arn)
 
     return extract_result(result, None, device)
+
+
+def estimate_cost_single_job(
+    job: Job, hybrid_iterations: int = 1, estimated_time: int = 3
+) -> float:
+
+    if TYPE_CHECKING:
```

Comment:
here I would put an actual guardrail in case user try to call this function for non `AWSDevice`s, something like
```py
is not isinstance(job.device, AWSDevice):
    raise JobDeviceMismatchError(...)
```